src/mpi/romio/adio/ad_lustre/ad_lustre_aggregate.c

```diff
@@ -311,3 +311,10 @@ int ADIOI_LUSTRE_Docollect(ADIO_File fd, int contig_access_count,
 
     return docollect;
 }
+
+void ADIOI_LUSTRE_Free_my_req(int nprocs, int *count_my_req_per_proc, ADIOI_Access * my_req)
+{
+    ADIOI_Free(count_my_req_per_proc);
+    ADIOI_Free(my_req[0].offsets);
```

Comment:
Looking at ADIOI_LUSTRE_Calc_my_req() I don't believe my_req[0].offsets holds the starting address of a block that can be freed; does it? See also my comment about buf[0].idx in ad_lustre_wrcoll.c

203     ptr += memLen + nprocs;     /* "+ nprocs" puts a terminal index at the end */
...
208     my_req[0].offsets = ptr;

Reply:
You are right. Looks like we have to add `buf_idx` into the `..._Free_my_req` call.

Reply:
@ashwinraghu Added two additional fixup commits. Can you review it again?

Reply:
Build broken:
../../../../src/mpi/romio/adio/ad_gpfs/ad_gpfs_rdcoll.c: In function 'ADIOI_GPFS_ReadStridedColl':
../../../../src/mpi/romio/adio/ad_gpfs/ad_gpfs_rdcoll.c:433:9: error: too few arguments to function 'ADIOI_Free_my_req'
  433 |         ADIOI_Free_my_req(nprocs, count_my_req_per_proc, my_req);
      |         ^~~~~~~~~~~~~~~~~
In file included from ../../../../src/mpi/romio/adio/include/adio.h:509,
                 from ../../../../src/mpi/romio/adio/ad_gpfs/ad_gpfs_rdcoll.c:16:
../../../../src/mpi/romio/adio/include/adioi.h:566:6: note: declared here
  566 | void ADIOI_Free_my_req(int nprocs, int *count_my_req_per_proc,
      |      ^~~~~~~~~~~~~~~~~

Reply:
Sorry. Just added the fixup. Try it again?

Reply:
More build errors:
../../../../src/mpi/romio/adio/ad_lustre/ad_lustre_aggregate.c:321:54: error: unknown type name 'ADIOI_Offset'; did you mean 'ADIO_Offset'?
  321 |                               ADIOI_Access * my_req, ADIOI_Offset ** buf_idx)
      |                                                      ^~~~~~~~~~~~
      |                                                      ADIO_Offset

Reply:
Yep, one more fixup. Try it again?

Reply:
Built alright. Running another round of tests just to be sure.

Reply:
The 1024-rank test ran fine. 
Note though that my tests do not cover changes under ad_lustre.

Reply:
Thanks @ashwinraghu.

Reply:
Okay. I'll look forward to seeing the merge on master after it gets approved.